server/jobs/storage_backend.go

```diff
@@ -11,12 +16,112 @@ type StorageBackend interface {
 	Read(key string) ([]string, error)
 
 	// Write logs to the storage backend
-	Write(key string, logs []string) (success bool, err error)
+	Write(key string, reader io.Reader) (success bool, err error)
+}
+
+type storageBackend struct {
+	location      stow.Location
+	logger        logging.SimpleLogging
+	containerName string
+}
+
+func (s *storageBackend) Read(key string) ([]string, error) {
+
+	logs := []string{}
+	err := stow.WalkContainers(s.location, s.containerName, 100, func(container stow.Container, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Skip if not right container
+		if container.Name() != s.containerName {
+			return nil
+		}
+
+		err = stow.Walk(container, key, 100, func(item stow.Item, err error) error {
+			if err != nil {
+				return err
+			}
+
+			// Found the right object
+			if item.Name() == key {
+
+				r, err := item.Open()
+				if err != nil {
+					return err
+				}
+
+				buf := new(strings.Builder)
+				_, err = io.Copy(buf, r)
+				if err != nil {
+					return err
+				}
+
+				logs = strings.Split(buf.String(), "\n")
+				return nil
```

Comment:
Since reading from the storage backend has a task of its own, I have removed the Read method implementation for now! I have addressed your comments and will push the changes in a later PR. I added this for the demo and forgot to remove it lol